tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java

```diff
@@ -308,9 +308,11 @@ public static void runEventLoop(Namespace parsedArgs) {
 
         String consumerGroup = parsedArgs.getString("consumerGroup");
 
-        final KafkaProducer<String, String> producer = createProducer(parsedArgs);
+        KafkaProducer<String, String> producer = createProducer(parsedArgs);
         final KafkaConsumer<String, String> consumer = createConsumer(parsedArgs);
 
+        int producerNumber = 0;
```

Comment:
```suggestion
        int producerNumber;
```
imho, its kind of the same right? idk which is the default, but again just offtopic im sorry.

This could be streamlined as well by: 

https://docs.openrewrite.org/recipes/staticanalysis/commonstaticanalysis

Reply:
this doesn't work for local variables, only class variables. get this error 
```
error: variable producerNumber might not have been initialized parsedArgs.getAttrs().put("transactionalId", parsedArgs.getString("transactionalId") + producerNumber++);
```

Reply:
ok thank you, my bad.